src/generic_core/globals.ts

```diff
@@ -56,6 +56,7 @@ export var settings :uproxy_core_api.GlobalSettings = {
   proxyBypass: DEFAULT_PROXY_BYPASS.slice(0),
   enforceProxyServerValidity: false,
   validProxyServers: [],
+  promo: null,
```

Comment:
Sorry, let me rephrase.

Why is it that we are saving a semi-random string instead of saving the actual promo code that the users are going to enter? (if we're handing them customized pieces of paper already)